quintagroup/transmogrifier/adapters/exporting.py

```diff
@@ -24,38 +29,60 @@ def __init__(self, context, transmogrifier):
         self.transmogrifier = transmogrifier
 
     def __call__(self, data):
-        data['data'] = self.exportReferences(data['data'])
+        data['data'] = self.exportReferences(safe_decode(data['data']))
         return data
 
     def exportReferences(self, xml):
         """ Marshall 1.0.0 doesn't export references, do it manually.
         """
         if xml is None:
-            return u"";
-        doc = minidom.parseString(xml)
-        root = doc.documentElement
-        for fname in self.context.Schema().keys():
-            if not isinstance(self.context.Schema()[fname], atapi.ReferenceField):
-                continue
-            values = self.context[fname]
-            if not values:
-                continue
-            elem = doc.createElement("field")
-            attr = doc.createAttribute("name")
-            attr.value = fname
-            elem.setAttributeNode(attr)
-            if type(values) not in (tuple, list):
-                values = [values,]
-            for value in values:
-                ref = doc.createElement('reference')
-                uid = doc.createElement('uid')
-                value = doc.createTextNode(str(value))
-                uid.appendChild(value)
-                ref.appendChild(uid)
-                elem.appendChild(ref)
-            root.appendChild(elem)
+            return u""
+        try:
+            doc = minidom.parseString(xml)
+        except ExpatError as e:
+            print '-->' * 26
```

Comment:
Ok, `print` is not the ultimate solution, and it could be improved. It won't hurt doctests in this case because it is done for ExpatErrors only (unless there are doctests for such errors).

I had a very hard time to find out where the error occurred which made my export fail; and I re-raise the exception. The default output of ExpatErrors is not sufficiently helpful. Having useful information (which enabled me to tweak my export to make it work) in foreground sessions only is better than having it not at all.